src/firewall/core/nftables.py

```diff
@@ -1618,10 +1620,10 @@ def build_set_create_rules(self, name, type, options=None):
                 break
 
         if options:
-            if "timeout" in options:
-                set_dict["timeout"] = options["timeout"]
-            if "maxelem" in options:
-                set_dict["size"] = options["maxelem"]
+            firewall.functions.dict_add_number(
+                set_dict, "timeout", options.get("timeout")
+            )
+            firewall.functions.dict_add_number(set_dict, "size", options.get("maxelem"))
```

Comment:
Could always do `int()`. `dict_add_number()` isn't adding any value here. The return value is not being checked. It silently fails if the value is None. I'd rather let it fail loudly. The type/values should be checked long before we get to the code path.

```
            if "timeout" in options:
                set_dict["timeout"] = int(options["timeout"])
            if "maxelem" in options:
                set_dict["size"] = int(options["maxelem"])
```

Reply:
changed.